### DISCUSSION TITLE
ppocr INFO: best metric, hmean: 0, is_float16: False

### DISCUSSION THREAD

**Discussion by xiaozuxin** [PaddlePaddle/PaddleOCR#12306] [Q&A]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：Windows 10
- 版本号/Version：
- Paddle：2.6  
- PaddleOCR：2.7  
- 问题相关组件/Related components：ch_PP-OCRv4_det_server_infer
- 运行指令/Command Code： python tools/train.py -c ./configs/rec/ch_ppocr_v2.0/rec_chinese_common_train_v2.0.yml
- 完整报错/Complete Error Message：ppocr INFO: best metric, hmean: 0, is_float16: False

各位大神，有没有遇到过这样的情况，windows10 下进行自己的推断训练时，
一直显示  ppocr INFO: best metric, hmean: 0, is_float16: False  是什么原因？
![ppocr](https://github.com/PaddlePaddle/PaddleOCR/assets/19402851/ecfe081e-b370-40c1-8f54-02a1de89a119)

**Comment by xiaozuxin** [comment]
```
Global:
  use_gpu: False
  epoch_num: 500
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/det_result/
  save_epoch_step: 1
  # evaluation is run every 5000 iterations after the 4000th iteration
  eval_batch_step: [1, 10]
  cal_metric_during_train: False
  #改为数据集路径
  pretrained_model: ./pretrain_models/ch_ppocr_server_v2.0_det_train/best_accuracy
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img: ./infer_img
  save_res_path: ./output/det_db/predicts_db.txt
  use_space_char: True

Architecture:
  model_type: det
  algorithm: DB
  Transform:
  Backbone:
    name: ResNet_vd
    layers: 18
    disable_se: True
  Neck:
    name: DBFPN
    out_channels: 256
  Head:
    name: DBHead
    k: 50

Loss:
  name: DBLoss
  balance_loss: true
  main_loss_type: DiceLoss
  alpha: 5
  beta: 10
  ohem_ratio: 3

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
    warmup_epoch: 2
  regularizer:
    name: 'L2'
    factor: 0

PostProcess:
  name: DBPostProcess
  thresh: 0.3
  box_thresh: 0.6
  max_candidates: 1000
  unclip_ratio: 1.5

Metric:
  name: DetMetric
  main_indicator: hmean

Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/
    label_file_list:
      - ./train_data/det/train.txt
    ratio_list: [1.0]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - DetLabelEncode: # Class handling label
      - IaaAugment:
          augmenter_args:
            - { 'type': Fliplr, 'args': { 'p': 0.5 } }
            - { 'type': Affine, 'args': { 'rotate': [-10, 10] } }
            - { 'type': Resize, 'args': { 'size': [0.5, 3] } }
      - EastRandomCropData:
          size: [960, 960]
          max_tries: 50
          keep_ratio: true
      - MakeBorderMap:
          shrink_ratio: 0.4
          thresh_min: 0.3
          thresh_max: 0.7
      - MakeShrinkMap:
          shrink_ratio: 0.4
          min_text_size: 8
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: ['image', 'threshold_map', 'threshold_mask', 'shrink_map', 'shrink_mask'] # the order of the dataloader list
  loader:
    shuffle: True
    drop_last: False
    batch_size_per_card: 8
    num_workers: 4

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/
    label_file_list:
      - ./train_data/det/val.txt
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - DetLabelEncode: # Class handling label
      - DetResizeForTest:
#           image_shape: [736, 1280]
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: ['image', 'shape', 'polys', 'ignore_tags']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1 # must be 1
    num_workers: 2

```

这是训练的模型yml

**Comment by Sunting78** [comment]
您好，是训练自己的数据集吗？训练过程中损失是否正常下降呢？

**Comment by xiaozuxin** [comment]
[train.log](https://github.com/PaddlePaddle/PaddleOCR/files/14943411/train.log)

@Sunting78   感谢，我是用  PPOCRLabel 标记自己的图片，划分数据集后，做的本地数据集训练，损失下降还不太会看，这是训练日志，能看到损失情况么。

**Comment by UserWangZz** [comment]
猜测是因为：
batch size = 8
train dataloader has 1 iters
这表示你的数据你数据集很小，只使用一个iteration，便可以遍历整个训练集
log_smooth_window : 20
表示log的平滑窗口是20个iteration，因此这可能就是训练时没有输出log信息的原因，以及训练效果不好的原因。
可以尝试加入更多的数据进行训练。
